get-resource.sh

```diff
@@ -28,10 +28,23 @@ else
     exit 1
 fi
 
+RHCOS_IMAGE_FILENAME_COMPRESSED=${RHCOS_IMAGE_FILENAME_QCOW/-openstack/-compressed}
 RHCOS_IMAGE_FILENAME_CACHED=cached-${RHCOS_IMAGE_FILENAME_QCOW}
 FFILENAME="rhcos-ootpa-latest.qcow2"
 
-mkdir -p /shared/html/images /shared/tmp
+mkdir -p /shared/html/images
+
+# Added to fix image path incompatibility between 4.7 and 4.8
+# as filed in https://bugzilla.redhat.com/show_bug.cgi?id=1972572
+if [[ -s "/shared/html/images/$RHCOS_IMAGE_FILENAME_QCOW/$RHCOS_IMAGE_FILENAME_COMPRESSED.md5sum" && ! -s "/shared/html/images/$RHCOS_IMAGE_FILENAME_QCOW/$RHCOS_IMAGE_FILENAME_CACHED.md5sum" ]]; then
+    cd /shared/html/images
+    mv "${RHCOS_IMAGE_FILENAME_QCOW}/$RHCOS_IMAGE_FILENAME_COMPRESSED.md5sum" "${RHCOS_IMAGE_FILENAME_QCOW}/$RHCOS_IMAGE_FILENAME_CACHED.md5sum"
+    mv "${RHCOS_IMAGE_FILENAME_QCOW}/$RHCOS_IMAGE_FILENAME_COMPRESSED" "${RHCOS_IMAGE_FILENAME_QCOW}/$RHCOS_IMAGE_FILENAME_CACHED"
```

Comment:
That is a good point, I think we have to discuss.

Reply:
I think these dual-stack problems started as a result of RHCOS moving from the network-legacy to network-manager dracut modules, which introduced a [slight change in behavior](https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/issues/729) meaning that the protocol chosen depends on a race between the two DHCP servers.

Looking at the [coreos config](https://github.com/coreos/fedora-coreos-config/commit/90ce41141f9349432a0621c6f0e00748e20c840f#diff-2c993c33dfed3829e59bfd9c90a811ef882875d7296adc5d28449b5d9db0f4d6) it seems this change happened several releases ago, but presumably the support status of dual-stack meant it wasn't noticed.  

@dustymabe can you confirm which RHCOS versions will be using NM in the ramdisk?

I think we may want to handle this on upgrade, and also backport the [kernel commandline fix](https://github.com/openshift/ironic-rhcos-downloader/pull/40) to previous releases, but that can probably be considered outside the scope of this PR, given that the commandline fix only currently exists on 4.8+ ?

/cc @sadasu @honza

Reply:
After sleep, it occurs to me that this PR is our best chance to fix this for those upgrading to 4.8, because at this point in the code we know we are dealing with a non-modified qcow. After the rename that happens here, we will have to go spelunking in the image to determine if it has been modified. (We're already [doing it twice](https://github.com/openshift/cluster-baremetal-operator/pull/163) in the image cache and that's bad enough.)

OTOH, we will need to do the spelunking thing in 4.7 anyway.

Reply:
> @dustymabe can you confirm which RHCOS versions will be using NM in the ramdisk?

4.6 and newer.